parlai/core/worlds.py

```diff
@@ -827,7 +827,7 @@ def create_task(opt, user_agents):
     # Single threaded or hogwild task creation (the latter creates multiple threads).
     # Check datatype for train, because we need to do single-threaded for
     # valid and test in order to guarantee exactly one epoch of training.
-    if opt.get('numthreads', 1) == 1 or 'train' not in opt['datatype']:
+    if opt.get('numthreads', 1) == 1 or 'train' not in opt['datatype'] or opt.get('batchsize', 1) > 1:
```

Comment:
maybe we need to explain the interplay of numthreads and batchsize here (as i dont understand myself)

Reply:
Basically, one can specify the amount of threads in the threadpool with numthreads; if they set a batchsize greater than 1, we should construct a BatchWorld rather than a HogwildWorld

Reply:
make comment a bit more explicit? also is it clear in parlai params?

Reply:
I'll make sure the comment is more explicit and that the parlai params are more clear in another PR